### PR TITLE
sys-apps/shadow: add 4.14.2

### PR DESCRIPTION
**Pull request by jubalh**

### CODE REVIEW

**Comment by thesamesam** [comment]
meh:
```
ERROR    OpenPGP verification failed for <_io.BufferedReader name='/var/tmp/portage/sys-apps/shadow-4.14.1/distdir/shadow-4.14.1.tar.xz'> (sig in
         /var/tmp/portage/sys-apps/shadow-4.14.1/distdir/shadow-4.14.1.tar.xz.asc):
         OpenPGP verification failed:
         gpg: Signature made Thu 05 Oct 2023 17:30:44 UTC
         gpg:                using RSA key EA3A87F0A4EBA030E45DF2409E8C1AFBBEFFDB32
         gpg: Can't check signature: No public key
````

I guess we need to add Alex's key to the keyring package too.

**Comment by jubalh** [comment]
> I guess we need to add Alex's key to the keyring package too.

Yeah we discussed on upstream IRC that we should publish this somewhere official.
Where would I have to add his key in Gentoo? I couldn't find it unfortunately.

**Comment by thesamesam** [comment]
Right now, we use `sec-keys/openpgp-keys-sergehallyn`. I suppose we'd have to vary the key package used depending on if we're on a stable branch (check for number of elements in the version? e.g. if [[ ${PV} == *.*.* ]] ; then ... fi; i can help with the ebuild mangling though if needed, just figuring out what the key is & where it's published is important).

So we'll need a new one for Alex.

**Comment by jubalh** [comment]
> So we'll need a new one for Alex.

He mentioned his key here: https://github.com/shadow-maint/shadow/pull/813#issuecomment-1737920648

>  i can help with the ebuild mangling though if needed

ok please do, I'm not sure I would make it right :)
I would have created one file with all keys of upstream shadow maintainers probably.

>  just figuring out what the key is & where it's published is important

After IRC, it is now discussed in upstream issue 815 about where to mention the official signing keys.

**Comment by thesamesam** [comment]
> > So we'll need a new one for Alex.
> 
> He mentioned his key here: [shadow-maint/shadow#813 (comment)](https://github.com/shadow-maint/shadow/pull/813#issuecomment-1737920648) (and in the next comment).
> 
> > i can help with the ebuild mangling though if needed
> 
> ok please do, I'm not sure I would make it right :) I would have created one file with all keys of upstream shadow maintainers probably.
> 

This wouldn't be wrong either, it just felt cleaner to do a new one for Alex given we already had a "personal" one for shadow, plus we might be able to then use it for man-pages too.

If it was from scratch, a group one for shadow wouldn't be a bad shout.

> > just figuring out what the key is & where it's published is important
> 
> After IRC, it is now discussed in upstream issue 815 about where to mention the official signing keys.

Thanks!

**Comment by thesamesam** [comment]
I'll wait until https://github.com/shadow-maint/shadow/issues/815 is sorted.

**Comment by jubalh** [comment]
They keys are now mentioned in the authors file.

**Comment by jubalh** [comment]
Added 4.14.2 as well

**Comment by floppym** [comment]
I see no point in adding both 4.14.1 and 4.14.2.

**Comment by jubalh** [comment]
So should I drop the first commit?

**Comment by floppym** [comment]
Yes please.

**Comment by jubalh** [comment]
Done

File: sys-apps/shadow/shadow-4.14.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,270 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+# Upstream sometimes pushes releases as pre-releases before marking them
+# official. Don't keyword the pre-releases!
+# Check https://github.com/shadow-maint/shadow/releases.
+
+VERIFY_SIG_OPENPGP_KEY_PATH="${BROOT}"/usr/share/openpgp-keys/sergehallyn.asc
```
Comment (floppym) [inline sys-apps/shadow/shadow-4.14.2.ebuild:10]: I think this needs to be updated with the new key?

Comment (jubalh) [reply sys-apps/shadow/shadow-4.14.2.ebuild:10]: Please see the comments above

Comment (floppym) [reply sys-apps/shadow/shadow-4.14.2.ebuild:10]: Oh, so this is waiting for @thesamesam to create a sec-keys ebuild?

Comment (jubalh) [reply sys-apps/shadow/shadow-4.14.2.ebuild:10]: I think @thesamesam is busy recently. But, yes, basically it's waiting for someone to create this :)
I'm not sure how to switch between the keys based on version. If I do it then I would creae a shadow-maintainers ebuild probably to not have to switch the keys :)

Comment (floppym) [reply sys-apps/shadow/shadow-4.14.2.ebuild:10]: Ok, I'll work on adding Alejandro's key.

**Comment by floppym** [comment]
See d73be0c445e0fc7919288a59169ee7dde3498b0c and b4759d375eb36ec1a0be3883e696f7c3323049fa for the new PGP key.

**Pull request closed without merging**